orchestrator-salt/src/main/resources/salt/salt/nginx/scripts/create-user-facing-cert.sh

```diff
@@ -2,7 +2,51 @@
 
 CERT_ROOT_PATH=/etc/certs-user-facing
 
-certm -d ${CERT_ROOT_PATH} ca generate -o=gateway --overwrite
-certm -d ${CERT_ROOT_PATH} server generate -o=gateway --host hostname --host {{ salt['pillar.get']('gateway:address') }}
+cat > ${CERT_ROOT_PATH}/openssl.cnf <<EOF
+[ req ]
+default_bits       = 2048
+default_md         = sha256
+prompt             = no
+encrypt_key        = no
+organizationName   = gateway
+distinguished_name = gateway_ca
+req_extensions     = v3_ca
+x509_extensions    = v3_ca
+
+[gateway_ca]
+organizationName    = gateway
+
+[gateway_server]
+organizationName    = gateway
+
+[ v3_ca ]
+basicConstraints        = critical,CA:TRUE
+subjectKeyIdentifier    = hash
+keyUsage                = critical,digitalSignature,keyEncipherment,keyAgreement,keyCertSign
+EOF
+
+cat > ${CERT_ROOT_PATH}/server_cert_ext.cnf <<EOF
+basicConstraints = critical,CA:FALSE
+keyUsage = critical,digitalSignature,keyEncipherment
+extendedKeyUsage = serverAuth,clientAuth
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS = hostname
+IP = {{ salt['pillar.get']('gateway:address') }}
```

Comment:
This needs to be a valid ip address. If this happens to be an fqdn then it will fail. I checked the code and I saw that it is an ip address. Please comment if this can be some other arbitrary string.

Reply:
AFAIK this has to be an IP but we don't have any validation when building `GatewayConfig`